design-system/react/src/components/InputAmount/InputAmount.tsx

```diff
@@ -76,7 +76,7 @@ export const InputAmount: InputAmountComponent = ({
       formatOpts.units
     );
     if (!currentAmount.eq(amount)) {
-      onChange?.(amount);
+      onChange?.(newText.length ? amount : undefined);
```

Comment:
@luizstacio so we should return null here instead?

Reply:
yes